src/worldcereal/rdm_api/rdm_interaction.py

```diff
@@ -0,0 +1,180 @@
+        A list of two strings representing the temporal extent, by default None. If None, all available data will be queried.
+    Returns
+    -------
+    List[str]
+        A List containing the URLs of all intersection collection IDs.
+    """
+
+    bbox = geometry.bounds
+    bbox_str = f"Bbox={bbox[0]}&Bbox={bbox[1]}&Bbox={bbox[2]}&Bbox={bbox[3]}"
+
+    val_time = (
+        f"&ValidityTime.Start={temporal_extent[0]}T00%3A00%3A00Z&ValidityTime.End={temporal_extent[1]}T00%3A00%3A00Z"
+        if temporal_extent
+        else ""
+    )
+
+    url = f"{RDM_ENDPOINT}/collections/search?{bbox_str}{val_time}"
+
+    response = requests.get(url)
+    response_json = response.json()
+
+    col_ids = []
+    for col in response_json:
+        col_ids.append(col["collectionId"])
+
+    return col_ids
+
+
+def _get_download_urls(collection_ids: List[str]) -> List[str]:
+    """Queries the RDM API and finds all HTTP URLs for the GeoParquet files for each collection ID.
+
+    Parameters
+    ----------
+    collection_ids : List[str]
+        A list of collection IDs.
+
+    Returns
+    -------
+    List[str]
+        A List containing the HTTPs URLs of the GeoParquet files for each collection ID.
+    """
+    urls = []
+    for id in collection_ids:
+        url = f"{RDM_ENDPOINT}/collections/{id}/download"
+        headers = {
+            "accept": "*/*",
+        }
+        response = requests.get(url, headers=headers)
+        urls.append(response.text)
+
+    return urls
+
+
+def _setup_sql_query(
+    urls: List[str],
+    geometry: BaseGeometry,
+    columns: List[str],
+    temporal_extent: Optional[List[str]] = None,
+) -> str:
+    """Sets up the SQL query for the GeoParquet files.
+
+    Parameters
+    ----------
+    urls : List[str]
+        A list of URLs of the GeoParquet files.
+    geometry : BaseGeometry
+        A user-defined geometry.
+    columns :
+        A list of column names to extract.
+    temporal_extent : Optional[List[str]], optional
+        A list of two strings representing the temporal extent, by default None. If None, all available data will be queried.
+
+    Returns
+    -------
+    str
+        A SQL query for the GeoParquet files.
+    """
+
+    combined_query = ""
+    columns_str = ", ".join(columns)
+
+    optional_temporal = (
+        f"AND valid_time BETWEEN '{temporal_extent[0]}' AND '{temporal_extent[1]}'"
+        if temporal_extent
+        else ""
+    )
+
+    for i, url in enumerate(urls):
+        query = f"""
+            SELECT {columns_str}, ST_AsWKB(ST_Intersection(geometry, ST_GeomFromText('{str(geometry)}'))) AS wkb_geometry
+            FROM read_parquet('{url}')
+            WHERE ST_Intersects(geometry, ST_GeomFromText('{str(geometry)}'))
+            {optional_temporal}
+
+        """
+        if i == 0:
+            combined_query = query
+        else:
+            combined_query += f" UNION ALL {query}"
+
+    return combined_query
+
+
+def query_ground_truth(
```

Comment:
I think it's more interesting to have this method return the `GeoDataFrame` so it can also be used without having to write the file to disk immediately. Another method can still do so if needed, but in my tests it's more useful to not write it to disk but keep in memory.

Reply:
Currently, right before the writing to parquet, there is also the 'sampling' part, which in this case is just taking the centroid:
```python
gdf.geometry = (
    gdf.geometry.centroid
)  # Ensure that the geometry is a point, can be replaced by a more advanced algorithm
```

Maybe we can also leave this out, so that a user can apply a more advanced sampling method if desired? 

We then just return a GeoDataFrame where all geometries are fully within the user-specified geometry. (Geometries which were only partially in have been cut-off).

Reply:
ah yes this is a good point I even overlooked. Yes because in my case I need the original geometries indeed.